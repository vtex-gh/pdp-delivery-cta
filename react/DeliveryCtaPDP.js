import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
