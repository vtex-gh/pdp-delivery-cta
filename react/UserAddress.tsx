import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
