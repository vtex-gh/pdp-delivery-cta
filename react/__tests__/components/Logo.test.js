import React from 'react'
import { render } from '@vtex/test-tools/react'

import Logo from '../../Logo'

describe('<Logo /> component', () => {
  const renderComponent = customProps => {
    const props = {
      title: 'title',
      ...customProps,
    }

    const comp = <Logo {...props} />

    return render(comp)
  }

  it('should match snapshot with link', () => {
    const { asFragment } = renderComponent({ href: 'http://logotest.test' })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot without link', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with url', () => {
    const { asFragment } = renderComponent({ url: 'http://logourl.test' })

    expect(asFragment()).toMatchSnapshot()
  })
})
