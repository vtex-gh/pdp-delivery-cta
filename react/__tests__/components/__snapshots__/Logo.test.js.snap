// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> component should match snapshot with link 1`] = `
<DocumentFragment>
  <a
    href="dummy"
  >
    <span
      class="store-logo logoContainer sizeDesktop pv4 ph6"
    >
      <svg
        fill="none"
        viewBox="0 0 142 80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          title
        </title>
        <rect
          fill="#F2F4F5"
          height="80"
          stroke="#E3E4E6"
          width="142"
        />
        <path
          d="M84.8156 47.5402L77.8157 33.9407C77.5479 33.4222 77.0282 33.0703 76.4349 33.0108C75.8434 32.9445 75.2572 33.1876 74.8844 33.6381L68.8854 40.9206L64.9707 38.386C64.5717 38.1293 64.08 38.0392 63.6162 38.1412C63.149 38.2432 62.7447 38.5271 62.4997 38.9249L57.2497 47.4246C56.9242 47.9499 56.9172 48.6026 57.227 49.1381C57.5367 49.6719 58.1195 50 58.7495 50H83.2494C83.8567 50 84.4184 49.694 84.7386 49.1925C85.0589 48.691 85.0851 48.0655 84.8156 47.5402Z"
          fill="#CACBCC"
        />
        <path
          d="M67.5 33C69.433 33 71 31.433 71 29.5C71 27.567 69.433 26 67.5 26C65.567 26 64 27.567 64 29.5C64 31.433 65.567 33 67.5 33Z"
          fill="#CACBCC"
        />
        <defs>
          <clippath
            id="clip0"
          >
            <rect
              fill="white"
              height="65"
              width="127"
            />
          </clippath>
        </defs>
      </svg>
    </span>
  </a>
</DocumentFragment>
`;

exports[`<Logo /> component should match snapshot with url 1`] = `
<DocumentFragment>
  <span
    class="store-logo logoContainer sizeDesktop pv4 ph6"
  >
    <img
      alt="title"
      class="logoImage"
      src="http://logourl.test"
    />
  </span>
</DocumentFragment>
`;

exports[`<Logo /> component should match snapshot without link 1`] = `
<DocumentFragment>
  <span
    class="store-logo logoContainer sizeDesktop pv4 ph6"
  >
    <svg
      fill="none"
      viewBox="0 0 142 80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        title
      </title>
      <rect
        fill="#F2F4F5"
        height="80"
        stroke="#E3E4E6"
        width="142"
      />
      <path
        d="M84.8156 47.5402L77.8157 33.9407C77.5479 33.4222 77.0282 33.0703 76.4349 33.0108C75.8434 32.9445 75.2572 33.1876 74.8844 33.6381L68.8854 40.9206L64.9707 38.386C64.5717 38.1293 64.08 38.0392 63.6162 38.1412C63.149 38.2432 62.7447 38.5271 62.4997 38.9249L57.2497 47.4246C56.9242 47.9499 56.9172 48.6026 57.227 49.1381C57.5367 49.6719 58.1195 50 58.7495 50H83.2494C83.8567 50 84.4184 49.694 84.7386 49.1925C85.0589 48.691 85.0851 48.0655 84.8156 47.5402Z"
        fill="#CACBCC"
      />
      <path
        d="M67.5 33C69.433 33 71 31.433 71 29.5C71 27.567 69.433 26 67.5 26C65.567 26 64 27.567 64 29.5C64 31.433 65.567 33 67.5 33Z"
        fill="#CACBCC"
      />
      <defs>
        <clippath
          id="clip0"
        >
          <rect
            fill="white"
            height="65"
            width="127"
          />
        </clippath>
      </defs>
    </svg>
  </span>
</DocumentFragment>
`;
